tests/integration/oup/test_oup_pull_ftp.py

```diff
@@ -21,48 +21,61 @@ def dag():
+    return dag.get_is_paused()
 
-def test_dag_run(dag: DAG):
-    repo = OUPRepository()
-    repo.delete_all()
-    assert len(repo.find_all()) == 0
-    id = datetime.datetime.utcnow().strftime("test_oup_pull_ftp_%Y-%m-%dT%H:%M:%S.%f")
-    dagrun = dag.create_dagrun(DagRunState.QUEUED, run_id=id)
-    wait().at_most(90, SECOND).until(lambda: get_dagrun_state(dagrun))
-    assert len(repo.find_all()) == 3
 
+class TestClassOUPPullFilesFromFTP:
+    def test_dag_loaded(self, dag: DAG):
+        assert dag is not None
+        assert len(dag.tasks) == 2
 
-def test_dag_migrate_from_FTP():
-    repo = OUPRepository()
-    repo.delete_all()
-    assert len(repo.find_all()) == 0
-    with OUPFTPService() as ftp:
-        migrate_from_ftp(
-            ftp,
-            repo,
-            get_logger().bind(class_name="test_logger"),
-            **{
-                "params": {
-                    "excluded_directories": [],
-                    "force_pull": False,
-                    "filenames_pull": {
-                        "enabled": False,
-                        "filenames": [],
-                        "force_from_ftp": False,
-                    },
-                }
-            },
+    def test_dag_run(
+        self,
+        dag: DAG,
+        dag_was_paused: bool,
+    ):
+        repo = OUPRepository()
+        repo.delete_all()
+        assert len(repo.find_all()) == 0
+        id = datetime.datetime.utcnow().strftime(
+            "test_oup_pull_ftp_%Y-%m-%dT%H:%M:%S.%f"
         )
+        if dag.get_is_paused:
+            DagModel.get_dagmodel(dag.dag_id).set_is_paused(is_paused=False)
+        dagrun = dag.create_dagrun(DagRunState.QUEUED, run_id=id)
+        wait().at_most(90, SECOND).until(lambda: self.get_dagrun_state(dagrun))
         assert len(repo.find_all()) == 3
+        if dag_was_paused:
+            DagModel.get_dagmodel(dag.dag_id).set_is_paused(is_paused=True)
 
+    def test_dag_migrate_from_FTP(self):
+        repo = OUPRepository()
+        repo.delete_all()
+        assert len(repo.find_all()) == 0
+        with OUPFTPService() as ftp:
+            migrate_from_ftp(
+                ftp,
+                repo,
+                get_logger().bind(class_name="test_logger"),
+                **{
+                    "params": {
+                        "excluded_directories": [],
+                        "force_pull": False,
+                        "filenames_pull": {
+                            "enabled": False,
+                            "filenames": [],
+                            "force_from_ftp": False,
+                        },
+                    }
+                },
+            )
+            assert len(repo.find_all()) == 3
 
-def get_dagrun_state(dagrun: DagRun):
-    dagrun.update_state()
-    return (
-        not dagrun.get_state() == DagRunState.QUEUED
-        and not dagrun.get_state() == DagRunState.RUNNING
-    )
+    def get_dagrun_state(self, dagrun: DagRun):
```

Comment:
maybe you can have this function in test utils instead of repeating it in all the classes?